src/db.js

```diff
@@ -10,22 +10,27 @@ if (flags.debug) {
 let db = null;
 
 const setConnect = () => {
-  const connect = mongoose.connect(
-    connectionDBString,
-    { useMongoClient: true },
-    (error) => {
-      if (error) {
-        logger.error(`Database connection can not be created:\n${error}`);
+  const promise = new Promise((resolve, reject) => {
+    const connect = mongoose.connect(
+      connectionDBString,
+      { useMongoClient: true },
+      (error) => {
+        if (error) {
+          logger.error(`Database connection can not be created:\n${error}`);
+        }
+        reject(error);
       }
-    }
-  );
-  return connect;
+    );
+    resolve(() => connect);
```

Comment:
I don't know why, but db promise has very strange behavior. It's work only if I return function. When I return a object from promise of "setConnect" I get `undefined`.

Reply:
Same behavior in async `connect` function. If I call `connect` like async function I get undefined (if I change return value from object to function all work fine). Now I did't have problem with it, because my app use this function not in another function and do it synchronously. I call `connect` in app.js like
```
import express from 'express';
import applyMiddleware from './middlewares';
import db from './db';

const app = express();

db.connect();

applyMiddleware(app);

export default app;

```
But I understand that this behavior doesn't right

Reply:
Let's deal with DB connection later then